test/remote/gateways/remote_simetrik_test.rb

```diff
@@ -13,51 +12,203 @@ def setup
     assert_success response
     assert_instance_of Response, response
-    assert_equal response.params['message'], 'successful authorize'
-    assert_equal response.params['code'][0], 'S', 'Should expected code like Sxxx'
-    assert_equal response.avs_result.to_hash['code'], 'G'
-    assert_equal response.cvv_result.to_hash['code'], 'P'
+    assert_equal response.message, 'successful authorize'
+    assert_equal response.error_code, nil, 'Should expected error code equal to nil'
+    assert_equal response.avs_result['code'], 'G'
+    assert_equal response.cvv_result['code'], 'P'
     assert response.test?
   end
 
   def test_failed_authorize
     response = @gateway.authorize(@amount, @credit_card, @authorize_capture_options_failed)
     assert_failure response
     assert_instance_of Response, response
-    # assert_equal response.message, 'Successful authorize'
-    assert_equal response.params['code'][0], 'R', 'Should expected code like Rxxx'
-    assert_equal response.avs_result.to_hash['code'], 'I'
-    assert_equal response.cvv_result.to_hash['code'], 'P'
+    assert_not_equal response.error_code, nil, 'Should expected error code not equal to nil'
+    assert_equal response.avs_result['code'], 'I'
+    assert_equal response.cvv_result['code'], 'P'
+    assert response.test?
+  end
+
+  def test_failed_authorize_by_invalid_card
+    response = @gateway.authorize(@amount, @credit_card_invalid, @authorize_capture_options_success)
+    assert_failure response
+    assert_instance_of Response, response
+    assert_not_equal response.error_code, nil, 'Should expected error code not equal to nil'
+    assert_equal response.avs_result['code'], 'G'
+    assert_equal response.cvv_result['code'], 'P'
     assert response.test?
   end
 
   def test_success_purchase
     response = @gateway.purchase(@amount, @credit_card, @authorize_capture_options_success)
     assert_success response
     assert_instance_of Response, response
-    assert_equal response.params['message'], 'successful charge'
-    assert_equal response.params['code'][0], 'S', 'Should expected code like Sxxx'
-    assert_equal response.avs_result.to_hash['code'], 'G'
-    assert_equal response.cvv_result.to_hash['code'], 'P'
+    assert_equal response.message, 'successful charge'
+    assert_equal response.error_code, nil, 'Should expected error code equal to nil'
+    assert_equal response.avs_result['code'], 'G'
+    assert_equal response.cvv_result['code'], 'P'
     assert response.test?
   end
 
   def test_failed_purchase
     response = @gateway.purchase(@amount, @credit_card, @authorize_capture_options_failed)
     assert_failure response
     assert_instance_of Response, response
-    assert_equal response.params['code'][0], 'R', 'Should expected code like Rxxx'
-    assert_equal response.avs_result.to_hash['code'], 'I'
-    assert_equal response.cvv_result.to_hash['code'], 'P'
+    assert_not_equal response.error_code, nil, 'Should expected error code not equal to nil'
```

Comment:
Instead of asserting that the `response.error_code` is not nil on these failure tests, can we assert the specific error code we expect? For example, `assert_equal response.error_code, 'config_error'`?

Reply:
We have changed the assertions in the latest commit as to assert the precise error code instead.